src/main/java/com/google/api/codegen/ProtocGeneratorMain.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen;
+
+import static com.google.api.codegen.ArtifactType.GAPIC_CODE;
+
+import com.google.api.codegen.gapic.GapicGeneratorApp;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.tools.ToolOptions;
+import com.google.api.tools.framework.tools.ToolUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import com.google.protobuf.compiler.PluginProtos;
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorRequest;
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorResponse;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Entrypoint for protoc-plugin invoked generation. Protoc passes input via std.in as a serialized
+ * CodeGeneratorRequest, and expects to read a CodeGeneratorResponse from std.out.
+ */
+public class ProtocGeneratorMain {
+
+  private static final ArtifactType DEFAULT_ARTIFACT_TYPE = GAPIC_CODE;
+
+  public static void main(String[] args) {
+    // TODO(andrealin): Implement. Handle I/O interfacing with protoc.
```

Comment:
Is the intention to submit this PR with this TODO here?

Reply:
I have another PR right on top of this one that will implement the main method. So yes, if that's ok.